### PR TITLE
Expect CancelledError when node is terminated

### PR DESCRIPTION
**Pull request by w1am**
Changed: Expect a `CancelledError` error when a node connection is abruptly terminated.

The default value of [ShutdownTimeout](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.kestrelserveroptions.shutdowntimeout?view=aspnetcore-1.1) in ESDB is now explicitly set to 5s, which was the default value for ESDB versions before v24.x. The value was recently changed to 30s in recent .NET releases, but this change was causing some issues with subscriptions. See more info [here](https://github.com/EventStore/EventStore/pull/4110).

### CODE REVIEW

File: src/__test__/extra/write-after-end.test.ts [hunk 1]
```diff
@@ -94,12 +94,7 @@ describe("write after end", () => {
 
       const error = await errorPromise;
 
-      // The reason for this discrepancy is explained in mid-stream.test.ts
-      if (matchServerVersion`<=23.10`) {
-        expect(error).toBeInstanceOf(CancelledError);
-      } else {
-        expect(error).toBeInstanceOf(UnavailableError);
-      }
+      expect(error).toBeInstanceOf(CancelledError);
```
Comment (George-Payne) [inline src/__test__/extra/write-after-end.test.ts:97]: Seems to still be consistently failing on next-feature-version:
image

Comment (w1am) [reply src/__test__/extra/write-after-end.test.ts:97]: It seems the error in the screenshot (line 31) does not match the original codebase (line 97). This behavior is new and unrelated to the previous release. It seems to be a timing sensitivity issue. I introduced some delay when killing a node.

File: src/__test__/utils/Cluster.ts [hunk 2]
```diff
@@ -267,6 +268,7 @@ export class Cluster {
     const response = await stopOne(nodeId, {
       cwd: this.path(),
     });
+    await delay(1000);
```
[the remark this answers is not in the document]
Comment (w1am) [reply src/__test__/utils/Cluster.ts]: Following our discussion:
- The "write after end" bug still exists.
- We will temporarily catch and throw a different error until this bug is resolved. See comment in the code

File: src/__test__/utils/Cluster.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ import type { EndPoint } from "../../types";
 
 import { testDebug } from "./debug";
 import { dockerImages } from "./dockerImages";
+import { delay } from "./delay";
```
Comment (George-Payne) [inline src/__test__/utils/Cluster.ts:14]: import not used

**Review by George-Payne — CHANGES_REQUESTED** [review]

File: src/__test__/utils/Cluster.ts (hunk 3 above)
Comment (George-Payne) [reply src/__test__/utils/Cluster.ts:14]: We should probably get ts to check for this, it's either:

this:
https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax

or maybe:
https://www.typescriptlang.org/tsconfig#noUnusedLocals

Comment (George-Payne) [reply src/__test__/utils/Cluster.ts:14]: (not in this PR)

File: src/__test__/extra/typedEvents-more.test.ts [hunk 4]
```diff
@@ -220,7 +220,6 @@ describe("typed events should compile", () => {
             confirmedAt: new Date(event.data.confirmedAt),
           };
         default: {
-          const _: never = event;
```
Comment (George-Payne) [inline src/__test__/extra/typedEvents-more.test.ts:223]: This isn't unused, it's an exhaustive check.

https://www.typescriptlang.org/docs/handbook/2/functions.html#never

Comment (w1am) [reply src/__test__/extra/typedEvents-more.test.ts]: I reset the history to where it was :)

**Review by George-Payne — APPROVED** [review]

**Pull request merged**